packages/utils/src/currency-formatter/README.md

```diff
@@ -0,0 +1,133 @@
+# Formatter
+
+Currency formatting utility that handles fiat and cryptocurrency display.
+
+`formatter` is a resilient wrapper around `Intl.NumberFormat` with baked in sensible defaults,
+presets, and few custom options.
+
+## Setup
+
+The formatter isn’t tied to our domain model and expects a numeric decimal input; therefore, it
+needs a one-time setup at the app-level to adapt the app’s currency to its input:
+
+```ts
+import { FormatAmountOptions, createCurrencyFormatter } from '@leather.io/utils';
+
+const formatter = createCurrencyFormatter({ locale: 'en-US' });
+
+function formatAmount(money: Money, options?: FormatAmountOptions) {
+  return formatter.formatAmount(
+    {
+      amount: money.amount.shiftedBy(-money.decimals).toNumber(),
+      currencyCode: money.symbol,
+      decimals: money.decimals,
+    },
+    options
+  );
+}
+```
+
+### Usage
+
+The formatter comes with sensible defaults that work for most scenarios.
+Defaults are resolution-dependent and slightly vary between cryptocurrency and fiat.
+
+```ts
+// Fiat currencies show currency symbols
+formatAmount(usdBalance); // "$1,234.56"
+formatAmount(jpyBalance); // "¥1,235" (no decimal places)
+formatAmount(eurBalance); // "1.234,56 €" (locale 'de')
+
+// Currency code for crypto is always appended regardless of locale
+formatAmount(btcBalance); // "12.3456789 BTC"
+formatAmount(ethBalance); // "456.123456 ETH"
+
+// Large amounts use compact notation (≥1M by default)
+formatAmount(millionDollarBalance); // "$1.23M"
+formatAmount(billionDollarBalance); // "$1.23B"
+
+// Small fiat amounts show approximation
+formatAmount(dustAmount); // "< $0.01"
+```
+
+### Presets
+
+Presets are a set of named configurations covering common formatting scenarios.
+While these scenarios can be easily achieved by configuring the formatter, presets serve a
+purpose of enforcing consistency across different apps.
+
+#### Price preset
+
+```ts
+// up to 6 decimals for small amounts 0-99
+formatAmount(smallPrice, { preset: 'token-price' }); // $0.000001
+formatAmount(mediumPrice, { preset: 'token-price' }); // $12.345679
+
+// 2 decimals for amounts ≥100
+formatAmount(largePrice, { preset: 'token-price' }); // $3,587.02
+formatAmount(btcPrice, { preset: 'token-price' }); // $122,838.46
+```
+
+#### Shorthand balances
+
+Meant to be used in listing items and other areas with scarce real estate.
+
+```ts
+formatAmount(btcBalance, { preset: 'shorthand-balance-crypto' }); // // "12.3457 BTC"
```

Comment:
The premise of `formatAmount` is to be currency-agnostic, so this isn't ideal.
I added ability to specify presets as callbacks to be input-aware after I added these presets, so these should be merged into a single `shorthand-balance` preset.